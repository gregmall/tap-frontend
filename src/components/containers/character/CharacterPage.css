.display{
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  width: 100%;
  margin: 10px;
  text-decoration: none;
  align-items: center;

  
}
p {
  text-align: center;
}
.card {
  padding:  10px;
  list-style: none;
  border-radius: 8px;
  margin: 5px;
  border: 1px solid #8142f5;
  background-color:white;
  color: #8142f5;
  text-decoration: none;
  height: auto;
  display: grid;
  text-align: center;
  align-items: center;
  justify-content: center;
  text-decoration: none;
 

  
  
}
.card img{
  
  width:15em;
  height:15em;
  border-radius: 6px;
  padding: 5px;
  background-color: #8142f5


}
.loading {
  display:flex;
  align-items: center;
  justify-content: center;
  margin-top: 20%;
}

