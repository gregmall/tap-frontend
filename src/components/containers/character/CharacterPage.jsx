import React from 'react';
import { Link } from 'react-router-dom';
import { useEffect } from 'react';
import { fetchCharacters } from '../../../actions/characterActions';

import { useCharacters } from '../../hooks/characters';
import styles from './CharacterPage.css';
import { useDispatch, useSelector } from 'react-redux';



const CharacterPage = () => {
  // const characters= useSelector(state => state.character.characters);
  const dispatch = useDispatch();
  const { loading, characters } = useCharacters();

  if(loading) return <span><img src="http://cdn.lowgif.com/full/ff8280aafe27319d-ajax-loading-gif-transparent-background-2-gif-images.gif"/></span>


 
    
 const characterElements = characters.map(character => (
  <Link to = {`/detail/${character.id}`}>
  <li li key={character.id} className={styles.item} >
    
  <p><h2>{character.name}</h2></p>
    <img src={character.image} />
    <p>{character.role}</p>
   

  </li>
  </Link>
  
));

  return (

    <ul data-testid="characters" className={styles.display}>
        {characterElements}
    </ul> 
)

}
export default CharacterPage;
