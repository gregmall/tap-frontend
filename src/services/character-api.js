const URL = process.env.REACT_APP_API_URL;

export const getCharacters = () => {
  return fetch(`${URL}/api/v1/characters`)
  .then(res => res.json())

  }


  // export const  getCharacters = async () =>{
  //   const res = await fetch(`${URL}/api/v1/characters`);
  //   const json = await res.json();
  //   console.log(json)
    
   
  
  //   return json.map(character => ({
  //     id: character.id,
  //     name: character.name,
  //     image: character.image,
  //     quote: character.quote,
  //     role: character.role
      
  //   }));
    
  // };
