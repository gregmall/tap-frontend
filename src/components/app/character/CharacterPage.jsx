import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { getCharacters } from '../../../services/character-api';


const CharacterList = () => {
  
 const characters= getCharacters();
    console.log(characters)
   



return (

  <div>hello</div>
)

}
export default CharacterList;
