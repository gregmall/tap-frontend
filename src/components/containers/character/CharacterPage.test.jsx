import React from 'react';
import { render } from '@testing-library/react';
import CharacterPage from './CharacterPage';
import { MemoryRouter, Route } from 'react-router-dom';
import { getCharacters } from '../../../services/character-api';

jest.mock('../../../services/character-api.js');
describe('Character Page component', () => {
  getCharacters.mockResolvedValue(
    {
      name: "Ian Faith",
        image: "https://pbs.twimg.com/media/Du0DfFwVYAEmLlc.jpg",
        quote: "Well, you should have seen the cover they WANTED to do! It wasn't a glove, believe me.",
        role: "Manager"
    },
    {
        name: "Nigel Tufnel",
        image: "https://miro.medium.com/max/676/0*THmfYKSeYsMtU2yi.jpg",
        quote: "This one goes to eleven.",
        role: "Lead guitar, mandolin"
    }
  );
  it('renders character component', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <Route path="/" component={CharacterPage}/>
      </MemoryRouter>
    );

    expect(asFragment()).toMatchSnapshot();

  });
});
