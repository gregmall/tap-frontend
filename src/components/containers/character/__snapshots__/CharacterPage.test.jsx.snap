// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character Page component renders character component 1`] = `
<DocumentFragment>
  <span>
    <img
      src="http://cdn.lowgif.com/full/ff8280aafe27319d-ajax-loading-gif-transparent-background-2-gif-images.gif"
    />
  </span>
</DocumentFragment>
`;
